util/timer_test.cc

```diff
@@ -15,122 +15,56 @@ class TimerTest : public testing::Test {
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->ClearAllCallBacks();
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
-        "InstrumentedCondVar::TimedWaitInternal", [&](void* arg) {
-          uint64_t* time_us = reinterpret_cast<uint64_t*>(arg);
-          if (*time_us < mock_env_->RealNowMicros()) {
-            *time_us = mock_env_->RealNowMicros() + 1000;
-          }
-        });
-    ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->EnableProcessing();
-  }
-#endif  // OS_MACOSX && !NDEBUG
-
   const uint64_t kSecond = 1000000;  // 1sec = 1000000us
+
+  void SetUp() override { mock_env_->InstallTimedWaitFixCallback(); }
 };
 
 TEST_F(TimerTest, SingleScheduleOnceTest) {
-  const int kIterations = 1;
-  uint64_t time_counter = 0;
-  mock_env_->set_current_time(0);
-
-  InstrumentedMutex mutex;
-  InstrumentedCondVar test_cv(&mutex);
-
+  const int kInitDelaySec = 1;
+  int mock_time_sec = 0;
+  mock_env_->set_current_time(mock_time_sec);
   Timer timer(mock_env_.get());
+
   int count = 0;
-  timer.Add(
-      [&] {
-        InstrumentedMutexLock l(&mutex);
-        count++;
-        if (count >= kIterations) {
-          test_cv.SignalAll();
-        }
-      },
-      "fn_sch_test", 1 * kSecond, 0);
+  timer.Add([&] { count++; }, "fn_sch_test", kInitDelaySec * kSecond, 0);
 
   ASSERT_TRUE(timer.Start());
 
+  ASSERT_EQ(0, count);
   // Wait for execution to finish
-  {
-    InstrumentedMutexLock l(&mutex);
-    while(count < kIterations) {
-      time_counter += kSecond;
-      mock_env_->set_current_time(time_counter);
-      test_cv.TimedWait(time_counter);
-    }
-  }
+  mock_time_sec += kInitDelaySec;
+  timer.TEST_WaitForRun([&] { mock_env_->set_current_time(mock_time_sec); });
 
   ASSERT_TRUE(timer.Shutdown());
-
   ASSERT_EQ(1, count);
 }
 
 TEST_F(TimerTest, MultipleScheduleOnceTest) {
-  const int kIterations = 1;
-  uint64_t time_counter = 0;
-  mock_env_->set_current_time(0);
-  InstrumentedMutex mutex1;
-  InstrumentedCondVar test_cv1(&mutex1);
-
+  const int kInitDelay1Sec = 1;
+  const int kInitDelay2Sec = 3;
+  int mock_time_sec = 0;
+  mock_env_->set_current_time(mock_time_sec);
   Timer timer(mock_env_.get());
+
   int count1 = 0;
-  timer.Add(
-      [&] {
-        InstrumentedMutexLock l(&mutex1);
-        count1++;
-        if (count1 >= kIterations) {
-          test_cv1.SignalAll();
-        }
-      },
-      "fn_sch_test1", 1 * kSecond, 0);
+  timer.Add([&] { count1++; }, "fn_sch_test1", kInitDelay1Sec * kSecond, 0);
 
-  InstrumentedMutex mutex2;
-  InstrumentedCondVar test_cv2(&mutex2);
   int count2 = 0;
-  timer.Add(
-      [&] {
-        InstrumentedMutexLock l(&mutex2);
-        count2 += 5;
-        if (count2 >= kIterations) {
-          test_cv2.SignalAll();
-        }
-      },
-      "fn_sch_test2", 3 * kSecond, 0);
+  timer.Add([&] { count2 += 5; }, "fn_sch_test2", kInitDelay2Sec * kSecond, 0);
```

Comment:
Is `5` picked randomly?

Reply:
Yes, I just switched it `count2++`, which is consistent with other tests.